File: 2351-first-letter-to-appear-twice/2351-first-letter-to-appear-twice.cpp
class Solution {
public:
    char repeatedCharacter(string s) {
        int s_set[26] = {0};
        
        for (auto &c:s) {
            if (s_set[c - 'a']) 
                return c;
            else
                ++s_set[c - 'a'];
        }
        
        return ' ';
    }
};

File: 2351-first-letter-to-appear-twice/2351-first-letter-to-appear-twice.rs
impl Solution {
    pub fn repeated_character(s: String) -> char {
        let mut sset = vec![0;26];
        
        for c in s.chars() {
            if sset[c as usize - 97] > 0 {
                return c;
            } else {
                sset[c as usize - 97] += 1;
            }
        }
        
        ' '
    }
}